src/extension.ts

```diff
@@ -34,43 +33,34 @@ class Counter {
 }
 
 export function activate(context: ExtensionContext) {
-    let serverOptions: ServerOptions;
+    window.setStatusBarMessage("RLS analysis: starting up"); 
 
-    let rls_root = process.env.RLS_ROOT;
-    window.setStatusBarMessage("RLS analysis: starting up");
+    let serverOptions: ServerOptions = () => new Promise<child_process.ChildProcess>((resolve, reject) => {
+        let rls_root = process.env.RLS_ROOT;
 
-    if (DEV_MODE) {
-        if (rls_root) {
-            serverOptions = {command: "cargo", args: ["run", "--release"], options: { cwd: rls_root } };
-        } else {
-            serverOptions = {command: "rustup", args: ["run", "nightly", "rls"]};
-        }
-    } else {
-        serverOptions = () => new Promise<child_process.ChildProcess>((resolve, reject) => {
-            function spawnServer(...args: string[]): child_process.ChildProcess {
-                let childProcess;
-                if (rls_root) {
-                    childProcess = child_process.spawn("cargo", ["run", "--release"], { cwd: rls_root });
+        function spawnServer(...args: string[]): child_process.ChildProcess {
+            let childProcess = rls_root ? child_process.spawn("cargo", ["run", "--release"], { cwd: rls_root })
+                                        : child_process.spawn("rustup", ["run", "nightly", "rls"]);
+
+            childProcess.on('error', err => {
+                if ((<any>err).code == "ENOENT") {
```

Comment:
Initializing `childProcess` with ternary properly infers a type for it, but because of that `tsc` expects `err` to be generic `Error` here, which is overriden in node. Couldn't find proper TS interface for that (`code` is specific to node here), so in this place I'm casting it to `any` (as it worked before) to avoid type checking.